packages/@aws-cdk/aws-dynamodb/lib/table-time-to-live-handler/index.ts

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable no-console */
+import type { IsCompleteRequest, IsCompleteResponse, OnEventRequest, OnEventResponse } from '@aws-cdk/custom-resources/lib/provider-framework/types';
+import { DynamoDB } from 'aws-sdk'; // eslint-disable-line import/no-extraneous-dependencies
+import { disableTimeToLive, enableTimeToLive, isTimeToLiveCorrect, isTimeToLiveStable } from './utils';
+
+const dynamodb = new DynamoDB({ apiVersion: '2012-08-10' });
+
+export async function onEventHandler(event: OnEventRequest): Promise<OnEventResponse> {
+  console.log('Event: %j', event);
+
+  if (event.RequestType === 'Create' || event.RequestType === 'Update') {
+    const currentTtl = await dynamodb.describeTimeToLive({
+      TableName: event.ResourceProperties.TableName,
+    }).promise();
+    console.log('Describe time to live: %j', currentTtl);
+
+    switch (currentTtl.TimeToLiveDescription?.TimeToLiveStatus) {
+      case 'ENABLING':
+        throw new Error('Can not change time to live while it is being enabled.');
+        break;
+      case 'DISABLING':
+        throw new Error('Can not change time to live while it is being disabled.');
+        break;
+      case 'ENABLED':
+        if (event.ResourceProperties.TimeToLiveAttribute !== currentTtl.TimeToLiveDescription.AttributeName) {
+          await disableTimeToLive(event);
+        }
+        break;
+      case 'DISABLED':
+        if (event.ResourceProperties.TimeToLiveAttribute) {
+          await enableTimeToLive(event);
+        }
+        break;
+    }
+
+    console.log('Update table: %j', event.ResourceProperties.TableName);
+    return { PhysicalResourceId: event.ResourceProperties.TableName };
+  }
+
+  return {};
```

Comment:
This only came to my mind now, but we could remove it, when the custom resource is removed. In this scenario, we also wouldn’t require the empty string trick. The remove of the resource could indirectly handle this for us. Right now, however I coded it with the assumption, that going back to the default should leave the TTL as is (like how CFN works).

But after thinking again about it, changing it to remove the TTL on deletion and removing the empty string trick might be better. The only downside is, that destroying a stack with a TTL will take considerably longer compared to right now.

Reply:
I think there are 2 separate cases here:

1. The DynamoDB Table is completely removed from the Stack. In this case, there's nothing to do in this custom resource.
2. The `timeToLiveAttribute` property was removed from the Table, but the Table is still kept in the Stack. In this case, this custom resource needs to disable the TTL on that Table.

Does this make sense?

Reply:
Yes, from a logical point of view I think I understand what you mean. However, its unclear for me how we would differentiate between the two cases inside the custom resource? Wouldn’t they appear identical, as the custom resource will always be removed first?

Reply:
Then perhaps simply disabling the TTL from the Custom Resource, always, will work for both cases?

Reply:
One way to differentiate the two cases is to [describe the stack](https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_DescribeStacks.html) and look at its status. If it's `DELETE_IN_PROGRESS` then you can skip disabling the TTL?

Reply:
Thats a great idea, I didn't think of the CloudFormation API. I have changed the logic so that the custom resource only deletes the TTL if the table is kept.